Initial commit

DenisKachan opened the pull request:
Basic app for text analysis

DenisKachan committed:
Initial commit

pasha2291 commented on fileForReading.txt:
```diff
@@ -0,0 +1,4 @@
+GitHub, Inc. is a platform and cloud-based service for software development and version control, allowing developers to
+store and manage their code. It uses Git software, providing the distributed version control of Git plus access control,
+bug tracking, software feature requests, task management, continuous integration, and wikis for every project.
+It is commonly used to host open source software development projects.
```
Этот файл лучше в resources положить, как и fileForWriting, а не держать в корне проекта.

pasha2291 commented on src/main/java/com/solvd/textAnalyzer/InnerFunctionality/TextReader/BaseReader.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.Scanner;
+
+@Log4j2
+public abstract class BaseReader {
```
Абстрактный класс используется если есть общий функционал у всех потомков его наследующих. Этот функционал обычно и выносят в абстрактный класс. Тут больше по смыслу подходит интерфейс BaseReader, т.к. у любого ридера будут методы читать / записать... не в зависимости от ого, какой источник (файл, консоль, база данных и тп), меняется лишь конкретная реализация.

pasha2291 commented on src/main/java/com/solvd/textAnalyzer/InnerFunctionality/TextReader/BaseReader.java:
```diff
@@ -0,0 +1,35 @@
+
+    public abstract BaseReader countAmountOfLetters() throws IOException;
+
+    public abstract BaseReader findACertainWord() throws IOException;
```
Принцип SOLID, где S = Single responsibility. От ридера ожидаешь чтение / записи и/или иных манипуляций с источником информации. Считать слова, заниматься поиском по тексту и тп = задача уже другого класса, скажем TextProcessor, или чего-то подобного.

pasha2291 commented on src/main/java/com/solvd/textAnalyzer/InnerFunctionality/TextReader/TextReaderFromConsole.java:
```diff
@@ -0,0 +1,78 @@
+public class TextReaderFromConsole extends BaseReader {
+
+    @Override
+    public TextReaderFromConsole countUniqueWords() throws IOException {
```
В продолжение предыдущего комментария:
При такой реализации у каждого потомка BaseReader будут свои методы работы с текстом. То есть мы для ридера из консоли реализуем countUniqueWords(), потом для ридера из файла реализуем тот же метод countUniqueWords(), если появится еще источник информации, например база данных - мы третий раз будем писать countUniqueWords() реализуя счет уникальных слов по тексту. Т.е. писать одну и ту же операцию энное количество раз только потому, что текст берется из разных источников. Это надо выносить в отдельный класс для работы с текстом и там реализовывать. И потом лишь использовать их для обработки полученного текста. 
С появлением новых источников текста / новых требуемых функций по работе с текстом поддержание такой архитектуры станет практически нереальной задачей.

pasha2291 commented on src/main/java/com/solvd/textAnalyzer/InnerFunctionality/TextReader/TextReaderFromConsole.java:
```diff
@@ -0,0 +1,78 @@
+        log.info("Enter any kind of text");
+        String textFromConsole = scanner.nextLine();
+        writer.write(textFromConsole);
+        String text = StringUtils.replaceAll(textFromConsole, "[^a-zA-Z0-9\\s]", "");
```
Вот тут мы удаляем все, что не подпадает под регулярку [^a-zA-Z0-9\\s], т.е. кириллицу мы в расчет не берем?

pasha2291 commented on src/main/java/com/solvd/textAnalyzer/Main.java:
```diff
@@ -0,0 +1,24 @@
+
+import java.io.IOException;
+
+public class Main {
```
Main обычно не оставляют. Что-нибудь типа Runner, или RunService.

DenisKachan committed:
1. Files for writing/reading moved to the resources
2. Cyrillic analyzing logic added
3. Reader classes renamed
4. Main class renameed
